src/time.cr

```diff
@@ -500,7 +500,7 @@ struct Time
     end
   end
 
-  def_at(end_of_year) { Time.new(year, 12, 31, 23, 59, 59, 999, kind: kind) }
+  def_at(end_of_year) { Time.new(year, 12, 31, 23, 59, 59, nanosecond: 999, kind: kind) }
```

Comment:
these `at_end_of_*` are not returning the nearest representable time before the next interval.
`nanosecond: 999_999_999` should probably be used instead of the old 999 milliseconds value.

I am not sure how this is playing when printing number with millisecond precision. It should truncate and not round to.

Reply:
You're totally right, thank you for this in-depth review! I fixed it